src/app/explorer/components/InfiniteScrollGuilds.tsx

```diff
@@ -0,0 +1,81 @@
+"use client";
+
+import { Button } from "@/components/ui/Button";
+import { Card } from "@/components/ui/Card";
+import { Skeleton } from "@/components/ui/Skeleton";
+import { useInfiniteQuery } from "@tanstack/react-query";
+import { useIntersection } from "foxact/use-intersection";
+import { useAtomValue } from "jotai";
+import { useCallback, useEffect } from "react";
+import { searchAtom } from "../atoms";
+import { getGuildSearch } from "../fetchers";
+import { GuildCard } from "./GuildCard";
+
+export const InfiniteScrollGuilds = () => {
+  const search = useAtomValue(searchAtom);
+  const { data, fetchNextPage, hasNextPage, isFetchingNextPage, isLoading } =
+    useInfiniteQuery({
+      queryKey: ["guilds", search || ""],
+      queryFn: getGuildSearch(search),
+      initialPageParam: 1,
+      enabled: search !== undefined,
+      getNextPageParam: (lastPage) =>
+        lastPage.total / lastPage.pageSize <= lastPage.page
+          ? undefined
+          : lastPage.page + 1,
+    });
+
+  const [setIntersection, isIntersected, resetIsIntersected] = useIntersection({
+    rootMargin: "700px",
+  });
+
+  useEffect(() => {
+    if (!isFetchingNextPage) {
+      resetIsIntersected();
+    }
+  }, [resetIsIntersected, isFetchingNextPage]);
+
+  useEffect(() => {
+    if (isFetchingNextPage) return;
+    if (isIntersected && hasNextPage) {
+      fetchNextPage();
+    }
+  }, [isIntersected, hasNextPage, isFetchingNextPage, fetchNextPage]);
+
+  const guilds = data?.pages.flatMap((page) => page.items) || [];
+
+  return (
+    <section className="grid gap-2">
+      <div className="grid gap-4 sm:grid-cols-2 lg:grid-cols-3">
+        {isLoading
+          ? Array.from({ length: 33 }, (_, i) => (
+              <Card key={i}>
+                <Skeleton className="size-full h-[114px]" />
```

Comment:
Let's use the detailed guild card skeleton which I sent to you, it looks better in my opinion 😉